client/Assets/Scripts/Battle.cs

```diff
@@ -647,7 +647,10 @@ long pastTime
             {
                 bountyCompletionManager.GenerateBountyCompletionPopUp();
                 bountyCompletionManager.UpdatePlayerCurrencies();
-                MainScreenManager.bountyReward = (int)GameServerConnectionManager.Instance.bountySelected.Reward.Amount;
+                if(GameServerConnectionManager.Instance.bountySelected != null)
```

Comment:
```suggestion
                if(GameServerConnectionManager.Instance.bountySelected)
```
Isn't it the same? just wondering, not mandatory comment